# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/BacktrackingAllocator.cpp
# Commit: 920da79f43fc
# Full Hash: 920da79f43fcf5802ebd5eb2bf610a18e821dacc
# Author: Andy Wingo <wingo@igalia.com>
# Date: 2020-03-19 03:53:34
# Regressor Bug: 1620197
# File Overlap Count: 1
# Description:
#   Bug 1620197 - Enable multiple results from WebAssembly functions r=lth
#   
#   This patch enables multi-value calls and returns, adding some tests, and
#   conditionally disabling a couple expect-fail reftests that now pass.
#   
# ==============================================================================

diff -r 98d7f9e4839f -r 920da79f43fc js/src/jit/BacktrackingAllocator.cpp
--- a/js/src/jit/BacktrackingAllocator.cpp	Wed Mar 18 17:12:14 2020 +0000
+++ b/js/src/jit/BacktrackingAllocator.cpp	Wed Mar 18 16:36:45 2020 +0000
@@ -1186,7 +1186,7 @@
     const LUse* use = ins->getOperand(0)->toUse();
     VirtualRegister& area = vregs[use->virtualRegister()];
     const LStackArea* areaAlloc = area.def()->output()->toStackArea();
-    reg.def()->setOutput(areaAlloc->resultAlloc(ins));
+    reg.def()->setOutput(areaAlloc->resultAlloc(ins, reg.def()));
   }
 }
 